helix-term/src/commands.rs

```diff
@@ -5115,11 +5115,25 @@ fn surround_replace(cx: &mut Context) {
             }
         };
 
+        // Visual feedback
+        let selection = selection.clone();
+        let mut ranges: SmallVec<[Range; 1]> = SmallVec::new();
+        change_pos.chunks(2).for_each(|p| {
+            if p.len() == 2 {
+                let from = *p.first().unwrap();
+                let to = *p.get(1).unwrap();
+                ranges.push(Range::new(from, from + 1));
+                ranges.push(Range::new(to, to + 1));
+            }
+        });
+        let feedback = Selection::new(ranges, 0);
+        doc.set_selection(view.id, feedback);
+
         cx.on_next_key(move |cx, event| {
```

Comment:
What happens if esc is pressed? I think this next key operation then doesn't run

Reply:
Yes, it goes via the `None` branch, which is where I restore the original selection.